src/main/java/net/atos/zac/app/taken/TakenRESTService.java

```diff
@@ -186,16 +188,18 @@ public void vrijgeven(final RESTTaakVerdelenGegevens restTaakVerdelenGegevens) {
     @PATCH
     @Path("assign")
     public void assignTaak(final RESTTaak restTaak) {
-        assertPolicy(policyService.readTaakActies(taskService.readTask(restTaak.id)).getWijzigenToekenning());
+        final TaskInfo task = taskService.readTask(restTaak.id);
+        assertPolicy(taskService.getTaakStatus(task) != AFGEROND && policyService.readTaakActies(task).getToekennen());
         assignTaak(restTaak.id, restTaak.behandelaar != null ? restTaak.behandelaar.id : null, restTaak.zaakUuid);
     }
 
     @PATCH
     @Path("assignTologgedOnUser")
     public RESTTaak assignToLoggedOnUser(final RESTTaakToekennenGegevens restTaakToekennenGegevens) {
-        assertPolicy(policyService.readWerklijstActies().getZakenTaken() && policyService.readTaakActies(taskService.readTask(restTaakToekennenGegevens.taakId))
-                .getWijzigenToekenning());
-        final Task task = assignTaak(restTaakToekennenGegevens.taakId, loggedInUserInstance.get().getId(), restTaakToekennenGegevens.zaakUuid);
+        TaskInfo task = taskService.readTask(restTaakToekennenGegevens.taakId);
+        assertPolicy(taskService.getTaakStatus(task) != AFGEROND && policyService.readWerklijstActies().getZakenTaken() &&
```

Comment:
is het controleren op policyService.readWerklijstActies().getZakenTaken() echt nodig?